factiondraft.ttslua

```diff
@@ -0,0 +1,807 @@
+-- this script should be placed on a standard cube object with a color tint with alpha = 0
+
+
+local testPlayers = false
+local testPlayerCount = 6
+
+local ignoreReach = false
+
+local allDice = {}
+
+local playmats = {}
+
+local factions = {
+   { name = "Marquise de Cat", color = "Orange", reachVal = 10, setupName = "Marquise"},
+   { name = "Underground Duchy", color = "Brown", reachVal = 8, setupName = "Duchy"},
+   { name = "Eyrie Dynasties", color = "Blue", reachVal = 7, setupName = "Eyrie"},
+   { name = "Vagabond", color = "White", reachVal = 5, setupName = "Vagabond"},
+   { name = "Riverfolk Company", color = "Teal", reachVal = 5, setupName = "Riverfolk"},
+   { name = "Woodland Alliance", color = "Green", reachVal = 3, setupName = "Woodland"},
+   { name = "Corvid Conspiracy", color = "Purple", reachVal = 3, setupName = "Corvid"},
+   { name = "Vagabond (Second)", color = "Red", reachVal = 2, setupName = "Vagabond"},
+   { name = "Lizard Cult", color = "Yellow", reachVal = 2, setupName = "Lizard"},
+}
+
+local reachMins = {
+   [1] = 0, -- for testing
+   [2] = 17,
+   [3] = 18,
+   [4] = 21,
+   [5] = 25,
+   [6] = 28,
+}
+
+local selections = {}
+local selectionOrder = {}
+local turnOrder = {}
+local currentReach = 0
+
+local xmlDefaults = {
+   tag = "Defaults",
+   children = {
+      {
+         tag = "Button",
+         attributes = {
+            textColor = "#dcdccc",
+            color = "#1c1c1c",
+            fontSize = "90",
+         }
+      },
+      {
+         tag = "Text",
+         attributes = {
+            color = "#dcdccc",
+            fontSize = "90",
+         }
+      }
+   }
+}
+
+local xmlPanelStart = {
+   tag = "Panel",
+   attributes = {
+      id = "start",
+      width = "900",
+      height = "800",
+      showAnimation = "Grow",
+      hideAnimation = "Shrink",
+   },
+   children = {
+      {
+         tag = "VerticalLayout",
+         attributes = {
+            spacing = "7"
+         },
+         children = {
+            { tag = "Button", attributes = {text = "Start Draft", onClick="start"}},
+            { tag = "Button", attributes = {text = "Start Draft\n(Ignore Reach)", onClick="start(noreach)"}},
+         }
+      }
+   }
+}
```

Comment:
Setting local in global scope is useless :D
It;s not really a problem, but unnessacery

Reply:
Not strictly useless. Other objects can access non local variables at the script level, local variables cannot be accessed/changed outside the script